utils/s2n_result.h

```diff
@@ -15,6 +15,69 @@
 
 #pragma once
 
+/*
+ * The goal of s2n_result is to provide a strongly-typed error
+ * signal value, which provides the compiler with enough information
```

Comment:
Nit: I think we generally put big comments like this after the includes, rather than before? But idk, we don't have many examples :)

Reply:
not sure i agree with that convention. Why wouldn't the module-level documentation be the first thing, since it's the first thing you read to understand why the module exists? Why would includes be first?